packages/ledger/src/v2.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  Chain,
+  WalletInit,
+  EIP1193Provider,
+  ProviderAccounts,
+  createEIP1193Provider
+} from '@web3-onboard/common'
+import type {
+  EthereumProvider as LedgerEthereumProvider
+} from '@ledgerhq/connect-kit-loader'
+import { isHexString, LedgerOptionsWCv2 } from './index.js'
+import { JQueryStyleEventEmitter } from 'rxjs/internal/observable/fromEvent'
+
+// methods that require user interaction
+const defaultOptionalMethods = [
+  'eth_sendTransaction',
+  'eth_signTransaction',
+  'personal_sign',
+  'eth_sign',
+  'eth_signTypedData',
+  'eth_signTypedData_v4'
+]
+
+function ledger(options?: LedgerOptionsWCv2): WalletInit {
+  const projectId = options?.walletConnectVersion == 2 ? options.projectId : undefined
+  if (!projectId) {
+    throw new Error(
+      'WalletConnect requires a projectId. Please visit https://cloud.walletconnect.com to get one.'
+    )
+  }
```

Comment:
```suggestion
  if (!options?.projectId) {
    throw new Error(
      'WalletConnect requires a projectId. Please visit https://cloud.walletconnect.com to get one.'
    )
  }
```

Since we know that this will only be called for version 2, can we simplify to just check for the `projectId`?

Reply:
Yes, fixed.